### ISSUE TITLE
Feature Request: make calendar colours in the calendarpanel customizable

### ISSUE THREAD

**Issue by Fuchs** [home-assistant/frontend#5975]
## The request

Currently it seems that calendar colours are decided at random. 
It would be nice if a custom calendar colour could be set, even if it has to be done manually in the customize yaml for the calendar object. After a quick glance at the code I tried to set backgroundColor on the calendar object, which seems to have no effect.

## The alternatives

Alternatively it would be nice if colours specified by the calendar (e.g. caldav allows calendars to have a certain colour) would be used.

## Additional information

Home Assistant 0.110.1, calendars fetched via CalDAV

**Comment by bverkron** [comment]
Yes please! Having them arbitrarily chosen and not matching what I have in Google Calendar or iCloud is very annoying and confusing for the family. At minimum using the same colours defined in Google would be great. I *think* that's possible as IIRC even back on older version of iOS it could automatically use the same calendar colours as were pulled from the Google Calendar when setup.

**Comment by zsarnett** [comment]
This should be done on the entity registry probably and made available for more than just the calendar entities

**Comment by bverkron** [comment]
Are there other entities that have colours associated with them?

**Comment by zsarnett** [comment]
I don't believe so, yet. We talked today and so far adding it to the entity attributes seems to be the best option.

**Comment by bverkron** [comment]
FYI for users in the meantime. Seems the calendars added to the calendar card entity are always assigned colours in the same order.

Example: 
```
1st = red
2nd = green
3rd = blue
4th = purple
etc...
```

So you can kind of hack around it by ordering the calendars in the card to match your color preferences. Assuming you have enough of them and can line them up the way you want. Some dummy calendars might be needed in between to increment you to the next desired colour if that makes sense. Something to keep in mind in the interim. YMMV.
